### DISCUSSION TITLE
Health check fails behind a reverse proxy

### DISCUSSION THREAD

**Discussion by LeVraiRoiDHyrule** [qdm12/ddns-updater#726] [General]
Hi,
I am hosting ddns-updater behind nginx. I am using to update the IP on my Cloudflare DNS. I don't want to use Cloudflare proxy feature.

If I turn the `proxied` setting to on, healthcheck of the container succeed but upon updating, ddns-updater turns on the "Proxied" setting in Cloudflare.

If I turn the `proxied` setting to off, then healthcheck fails with the following error:
```
ERROR program is unhealthy: lookup IP addresses do not match
```
ddns-updater successfully updates the IP but is not able to check if it has been correctly updated, thus updating continuously.

What could lead to this error ?
Thanks in advance for any answer.

**Comment by qdm12** [comment]
Let's say your domain is `domain.com`.

The healthcheck, when the proxied option is false, resolves `domain.com` and compares the ip address with the ip address sent in the last update, if it doesn't match the healthcheck fails.
If the proxied option is true, it just checks your public ip address matches the last ip address sent for an update (since the domain.com resolves to an ip address belonging to Cloudflare. Not really what you want, especially because you don't want your domain proxied.

Now this error `program is unhealthy: lookup IP addresses do not match` can happen for up to the TTL of your record, because the resolvers cache the result. Does it keep on happening forever? How about the IP address, is it updated correctly if you check on the cloudflare website? Also what do you get with for example `nslookup domain.com`, do you get your public ip address or an older IP address?
